### ISSUE TITLE
Cannot remove Cache-Control: no-store and Pragma: no-cache

### ISSUE THREAD

**Issue by rkkoszewski** [slimphp/Slim#2246]
Hi,
I'm trying to serve files by using PHP (Only logged in users can download them).

I'm trying to make caching to work, but Slim 3 seems to force some headers over my own. The two headers are Cache-Control and Pragma. I'm setting my own Cache-Control  a no-cache header + cache header is sent. I tried removing both by using withoutHeader('Pragma') and  withoutHeader('Cache-Control') but this is completely ignored.

I installed  https://www.slimframework.com/docs/features/caching.html, and I can define the new headers by using this middleware, but the browser is still not caching as it is still receiving the double headers saying it should not cache anything.

I'm not using any other middlewares than the ones that come by default with Slim 3.

How can I disable Slim 3 to prevent caching??

Greetings and thanks.

**Comment by TKDBB84** [comment]
I think there's something else at play in your codebase, I'm fairly certain slim isn't forcing a cache header.  To test this I spun up a very quick Slim only application from packagist slim/slim
```
require __DIR__ . '/vendor/autoload.php';
$app = new \Slim\App([]);
$app->get('/', function ($request, $response, $args){ 
	$body = new Slim\Http\Body(fopen('php://temp', 'r+'));
	$body->write('foo');
	return $response->withBody($body);
});
$app->run();
```

Which yields the following headers:

```
curl -vvv http://localhost:8000
* Rebuilt URL to: http://localhost:8000/
*   Trying ::1...
* Connected to localhost (::1) port 8000 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.41.0
> Host: localhost:8000
> Accept: */*
> 
< HTTP/1.1 200 OK
< Host: localhost:8000
< Connection: close
< X-Powered-By: PHP/7.0.19
< Content-Type: text/html; charset=UTF-8
< Content-Length: 3
< 
* Closing connection 0
foo
```

**Comment by akrabat** [comment]
@TKDBB84 is correct in that Slim doesn't add any headers other than Content-Type and Content-Length. I assume therefore that your web server is doing this.

**Comment by rkkoszewski** [comment]
Hi, thanks @TKDBB84 for the code snippet.

I tried it out and indeed the Cache-Control and Pragma headers were gone. And I was able to find out what is causing the issue, altough I have no idea how to fix it.

It seems that the PHP function "session_start()" is causing the headers to be created. Seems like here somebody has posted a similar question here: https://stackoverflow.com/questions/40320221/php-session-adding-headers-pragma-no-cache

Thank you again for helping. Greetings.
